bigtable-dataflow-parent/bigtable-hbase-dataflow/src/main/java/com/google/cloud/bigtable/dataflow/CloudBigtableIO.java

```diff
@@ -201,9 +207,11 @@ public static Coder getCoder(CoderType type) {
     private static final long serialVersionUID = 1L;
 
     @Override
-    public Result next(ResultScanner<Row> resultScanner) throws IOException {
+    public Result next(ResultScanner<Row> resultScanner, ByteKeyRangeTracker rangeTracker)
+        throws IOException {
       Row row = resultScanner.next();
-      if (row == null) {
+      if (row == null
```

Comment:
There is a way using `markDone` to give slightly better endgame progress updates to runners that support autoscaling/work rebalancing.

I would probably flip it around a bit for clarity:

```java
if (row != null && rangeTracker.tryReturnRecordAt(true, ByteStringUtil...)) {
   // Have record that is still in range for this reader.
   return Adapters.ROW_ADAPTER....
}

rangeTracker.markDone();
return null;
```